packages/core/src/components/TextArea/TextArea.tsx

```diff
@@ -34,12 +37,19 @@ const TextArea = forwardRef(
     const numRows = rows || DEFAULT_ROWS[size];
     const helpTextId = helpText && `${id}-help-text`;
 
+    const [characterCount, setCharacterCount] = useState(rest.value?.length || 0);
+
+    const handleOnChange = useCallback((event: React.ChangeEvent<HTMLTextAreaElement>) => {
+      setCharacterCount(event.target.value.length);
+      onChange?.(event);
+    }, [onChange]);
```

Comment:
Doing this as it seems like this component doesn't do any form of controlled input values, seemed overkill to implement it based on this feature.